src/main/java/com/o19s/es/ltr/query/RankerQuery.java

```diff
@@ -360,11 +363,11 @@ private void advanceSubIterators(int target) throws IOException {
             if (target == NO_MORE_DOCS) {
                 return;
             }
-            for (DocIdSetIterator iterator : subIterators) {
-                // FIXME: Probably inefficient
-                if (iterator.docID() < target) {
-                    iterator.advance(target);
-                }
+            while (subIteratorsPriorityQueue.top().iterator.docID() < target) {
```

Comment:
@nomoa  I am okay doing but wont this require a `downcasting` like so 

```java
((DisiWrapperWithFeatureOrdinal)iterator.subIteratorsPriorityQueue.top()).featureOrdinal
```
in order to access the featureOrdinal since the `disiPrioQueue.top()` interface returns the base class `DisiWrapper`.

Reply:
Indeed... do you think that casting will add too much overhead?

Reply:
Alternatively I wonder if there's way to make benefit of topList (see https://github.com/apache/lucene-solr/blob/4fb7e3d02c6354cf1a1dce522e652610f95bec24/lucene/sandbox/src/java/org/apache/lucene/search/CoveringScorer.java )

That should perhaps help us to need to cast only once?